libs/vkd3d/command.c

```diff
@@ -4883,10 +4883,10 @@ static void vk_access_and_stage_flags_from_d3d12_resource_state(const struct d3d
 
             case D3D12_RESOURCE_STATE_RENDER_TARGET:
                 *stages |= VK_PIPELINE_STAGE_2_COLOR_ATTACHMENT_OUTPUT_BIT;
-                /* If the corresponding image layout is COLOR_ATTACHMENT_OPTIMAL, we won't get automatic barriers,
+                /* If the corresponding image layout is COLOR_ATTACHMENT_OPTIMAL or GENERAL, we won't get automatic barriers,
                  * so add access masks as appropriate. */
-                if (d3d12_resource_pick_layout(resource, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL) ==
-                        VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL)
+                if ((d3d12_resource_pick_layout(resource, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL) ==
```

Comment:
I think it's more appropriate to just remove this branch altogether and always do the access flag. The "automatic" barrier thing seems like it's an outdated concept now. I think it's from the old legacy render pass days.

Reply:
Yes, it seems that direct assignment is also reasonable.